docs/adr/0002-yarn-v4.md

```diff
@@ -0,0 +1,175 @@
+# Extend Yarn Berry support to v4
+
+## Context
+
+Yarn v4 was released in October 23, 2023, which coincided with the time that the support for Yarn v3 was being introduced in Cachi2. To limit the scope of implementation, we decided to leave support for v4 to a later time. Now, as more projects migrate to v4, introducing proper support has become a priority.
+
+This document analyzes the the most impactful changes introduced with v4, and how they relate to the current v3 implementation.
+
+Main references:
+- [v4 blog post](https://yarnpkg.com/blog/release/4.0)
+- [Breaking changes for Yarn 4](https://github.com/yarnpkg/berry/issues/3591)
+
+### Global cache is enabled by default
+
+The [enableGlobalCache](https://yarnpkg.com/configuration/yarnrc#enableGlobalCache) option allows the user to set a shared location for the cache folder. Cachi2 already treats this option the following way:
+
+- Set to `true` during the prefetch, and point it to a specific location in order to keep the prefetched dependencies.
+- Set to `false` during the build, since the build needs to read from the cache, but the dependencies need to be installed to a local folder since they'll be used during the runtime.
+
+The only action that might be necessary is to document this behavior, since now during build-time, Cachi2 is setting the opposite option that would be expected as default.
+
+### All official plugins are enabled by default
+
+And can no longer be disabled by `.yarnrc.yml` configuration. The current official plugins don't seem to introduce any behavior that would taint the accuracy of the prefetched dependencies, though. Most of them only add support to protocols (which will still be filtered using the same rules of Yarn v4) or CLI commands.
+
+The list of all official plugins can be found under the "Default Plugins" section in the [API](https://yarnpkg.com/api) page on the official Yarn documentation. Here's a short summary of every official plugin (as of 4.5.3):
+
+<details>
+    <summary>Plugins that add support for a protocol</summary>
+
+- plugin-exec
```

Comment:
Can you refresh my mind on how the `exec` protocol cannot lead to arbitrary code execution in a transitive package? My memory has already faded.

Reply:
It can lead to arbitrary code execution, that's why we disallow it [here](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/package_managers/yarn/locators.py#L172-L176).

The reason for not disabling the plugin straight away is that it will make `yarn info` fail with an ugly error.